psutil/_psutil_posix.c

```diff
@@ -16,7 +16,7 @@
 #include <net/if.h>
 
 #ifdef PSUTIL_SUNOS10
-#include "arch/solaris10/ifaddrs.h"
+#include "arch/solaris/v10/ifaddrs.h"
```

Comment:
OpenSolaris is not the same with Solaris 5.10 x86. mib2.h in Solaris 5.10 x86 does not contain the extended information that sparc has, or OpenSolaris over mib2(the one that you reuqest with the define). At least not on my environment. So I do not think this works on Solaris x86. With the current code (using solaris10 as folder name) it shouldn't compile on Open Solaris, right? because that directory is not present. I have not tested on OpenSolaris.